apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/Emulator/metadata_server.py

```diff
@@ -0,0 +1,353 @@
+    elif isinstance(o, OrderedDict):
+      return self.fixupOrderedDict(o)
+
+    return o
+
+with open(os.path.join(os.path.dirname(os.path.realpath(__file__)), 'initial_data.json')) as initial_data_file:
+  all_data = json.load(initial_data_file, object_pairs_hook=OrderedDict)
+  indexed_lists = [
+    'instance/disks',
+    'instance/licenses',
+    'instance/network-interfaces',
+    'instance/network-interfaces/*/access-configs',
+    'instance/network-interfaces/*/forwarded-ips',
+    'instance/network-interfaces/*/ip-aliases'
+  ]
+  for indexed_list in indexed_lists:
+    components = indexed_list.split('/')
+    def process(data, i):
+      component = components[i]
+      if component == '*':
+        for item in data:
+          process(item, i + 1)
+      elif i == len(components) - 1:
+        data[component] = IndexedList(data[component])
+      else:
+        process(data[component], i + 1)
+
+    process(all_data, 0)
+
+not_found_error = formatResponse('Error: Not found', 404, common.content_type_html)
+bad_request_error = formatResponse('Error: bad request', 400, common.content_type_html)
+
+@app.after_request
+def add_headers(response):
+  response.headers[common.server] = common.server_value
+
+  # When the token is requested, the response should not have the Metadata-Flavor header
+  if not response.data.startswith('{"access_token":"') and not response.data.startswith('access_token '):
+    response.headers[common.metadata_flavor] = common.metadata_flavor_google
+  return response
+
+@app.route('/', methods = ['GET'])
+def root():
+  return formatResponse('0.1/\ncomputeMetadata/\n', 200, common.content_type_text)
+
+@app.route('/computeMetadata/', methods = ['GET'])
+def computeMetadataRoot():
+  return formatResponse('v1/\nv1beta1/\n', 200, common.content_type_text)
+
+@app.route('/computeMetadata/v1/', methods = ['GET'])
+def getRootData():
+  return getData('')
+
+@app.route('/computeMetadata/v1/<path:path>', methods = ['GET'])
+def getData(path):
+  # TODO There are 500 max hanging gets (not sure if we need to enforce this)
+  # TODO Actually handle waiting for changes and last_etag
+
+  def asBool(value):
+    return value.lower() in ['true', 'on', 'yes', '1']
+
+  def asInt(value, default):
+    try:
+      if value:
+        return int(value)
+      else:
+        return default
+    except ValueError:
+      return default
+
+  def convertToTextRecursive(prefix, o):
+    text = ''
+    if isinstance(o, list):
+      for val in o:
+        text += convertToTextRecursive(prefix, val)
+    elif isinstance(o, IndexedList):
+      for i in range(len(o)):
+        text += convertToTextRecursive(prefix + ('/' if len(prefix) != 0 else '') + unicode(i),
+                                       o[i])
+    elif isinstance(o, OrderedDict):
+      for key in o:
+        if excludeKeyInRecursiveGet(key):
+          continue
+        text += convertToTextRecursive(prefix + ('/' if len(prefix) != 0 else '') + key,
```

Comment:
I was originally just using normal strings but when I loaded data from the initial_data.json file and things were coming in as unicode values, I realized that str is not Unicode (sorry, I'm still very new to python) and so I changed all str uses to be unicode. But I just realized now I forgot to also prefix string literals with u. I will go through and do this. As for key here, it actually is a unicode value from what I can tell as it was also loaded from initial_data.json.